public/samples/on-chain-gas-estimate.sol

```diff
@@ -22,25 +36,28 @@ contract SenderReceiver is AxelarExecutable {
         string calldata message_
     ) external payable {
         bytes memory payload = abi.encode(message_);
-        
-        uint256 estimate = estimator.estimateGasFee(destinationChain, destinationAddress, payload, 200000, '');
 
+        uint256 estimate = getEstimate(
+            destinationChain,
+            destinationAddress,
+            payload
+        );
 
-        gasService.payGas{value: estimate} (
+        gasService.payGas{value: estimate}(
             address(this),
             destinationChain,
             destinationAddress,
             payload,
-            200000,
-            true,
-            msg.sender,
-            ''
+            gasLimit, // uint256 executionGasLimit - destination gas limit
+            true, // bool estimateOnChain - was this gas estimated on-chain?
```

Comment:
This boolean is to enforce checking supplied value against the estimate performed inside of the gas service. If payment is not enough it will revert